database/postStore.go

```diff
@@ -52,4 +62,23 @@ func (c *Conn) DeleteByID(id primitive.ObjectID) (*mongo.DeleteResult, error) {
 	return result, nil
 }
 
-func (c *Conn) DefineEndPoints() {}
+func (s *PostStore) ListAll() (*[]Post, error) {
+	postCollection := s.db.Collection(POST_COLL)
+	cursor, err := postCollection.Find(context.Background(), bson.M{})
+	if err != nil {
+		return nil, err
+	}
+	defer cursor.Close(context.Background())
+
+	var posts []Post
+	for cursor.Next(context.Background()) {
+		post := Post{}
+		err = cursor.Decode(&post)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I like the errors inlined (`if err := ...; err != nil ...`) but maybe it's just me.

I just realized we can't have democratic voting here because there's like, you know, two of us here. Happy coding!